mobile_payment/gql_mutations.py

```diff
@@ -0,0 +1,421 @@
+    token = data['policy']['token']
+    json_content = json.dumps(data.get('json_content', []))
+    
+    
+
+    try:
+        insuree = Insuree.objects.filter(chf_id=chf_id, validity_to__isnull=True).first()
+        if not insuree:
+            return {"success": False, "message": "Please pass in the correct CH ID."}
+    except Insuree.DoesNotExist:
+        return {"success": False, "message": "Please pass in the correct CHF ID."}
+
+    
+    policy = Policy.objects.filter(family=insuree.family, uuid=token).first()
+    if policy:
+        product = policy.product
+        if (
+            amount == product.lump_sum
+            and policy.status == Policy.STATUS_IDLE
+            and product_name == product.name
+            and product_code == product.code
+        ):
+            psp_service_provider = PaymentServiceProvider.objects.get(uuid=psp_service_provider_uuid)
+            transactions = Transactions.objects.create(amount=amount, payment_service_provider=psp_service_provider, status=1, json_content=json_content)
+            transactions.save()
+            premium = Premium.objects.create(amount=amount, policy=policy, pay_date=date.today(), pay_type="Mobile", transaction=transactions)
+            premium.save()
+            update_policy(premium, policy)
+            return {'success': True}
+        elif amount < product.lump_sum and policy.status == Policy.STATUS_IDLE and product_name == product.name and product_code == product.code:
+            return {'success': False, 'message': "Amount is too small. It needs to be the exact amount of the product."}
+        elif amount > product.lump_sum and policy.status == Policy.STATUS_IDLE and product_name == product.name and product_code == product.code:
+            return {'success': False, 'message': "Amount is too large. pls enter the excatt amount the product cost."}
+        elif amount == product.lump_sum and policy.status != Policy.STATUS_IDLE and product_name == product.name and product_code == product.code:
+            return {'success': False, 'message': "Please check again if your policy you are trying to pay for is idle  or active."}
+        else:
+            return {'success': False, 'message': "Please check again if your the policy is already active or has expired. Only policies with idle status can make payment at the moment."}
+    else:
+        return {"success": False, "message": "insuree does not have polciy contact NHIS for more information."}
+        
+def update_policy(premium, policy):
+    policy.status = Policy.STATUS_ACTIVE
+    policy.effective_date = premium.pay_date if premium.pay_date > policy.start_date else policy.start_date
+    policy.save()
+
+class ProcessPayment(graphene.Mutation):
+    class Arguments:
+        input_data = ProcessPaymentInput(required=True)
+
+    Output = graphene.String
+
+    @staticmethod
+    def mutate(root, info, input_data=None):
+        try:
+            if info.context.user is AnonymousUser:
```

Comment:
I understand that this mutation will be called directly by Afrimoney. For the sake of traceability, we need a technical user to identify it comes from Afrimoney.

Reply:
that's why interactive_user is added as foreign key in the PSP model to track which payment service provider is making a call from our API.